hydra-node/src/Hydra/Chain/Direct/Tx.hs

```diff
@@ -194,7 +197,7 @@ commitTx scriptRegistry networkId party utxo (initialInput, out, vkh) =
   initialScriptRef =
     fst (initialReference scriptRegistry)
   initialDatum =
-    mkScriptDatum $ Initial.datum ()
+    mkScriptDatum $ Initial.InitialDatum (headIdToCurrencySymbol headId)
```

Comment:
Probably because we are creating a `ScriptWitness` using this datum so the types would not align. Previously we used the helper function `datum` to encode empty datum but now we are using `InitialDatum` with the new field for currency symbol.